src/System.Net.Primitives/src/System/Net/SocketAddressPal.Unix.cs

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Sockets;
+using System.Text;
+
+namespace System.Net
+{
+    internal static class SocketAddressPal
+    {
+        public const int IPv6AddressSize = Interop.libc.sockaddr_in6.Size;
+        public const int IPv4AddressSize = Interop.libc.sockaddr_in.Size;
+        public const int DataOffset = 0;
+
+        public static unsafe AddressFamily GetAddressFamily(byte[] buffer)
+        {
+            fixed (byte* rawAddress = buffer)
```

Comment:
Presumably the buffer being passed in is known to be of an appropriate size that it can have its contents cast to a sockaddr?  Should we assert anything to help verify that?